src/actions/guides/http_and_routing/routing_and_params.cr

```diff
@@ -336,16 +336,100 @@ class Guides::HttpAndRouting::RoutingAndParams < GuideAction
 
     > Learn more about [error handling](#{Guides::HttpAndRouting::ErrorHandling.path}).
 
-    ## Query parameters
+    ## Handling parameters
 
-    Other times you may want to accept parameters in the query string, e.g. `https://example.com?page=2`.
+    Parameters, or `params`, are data that is sent from client back to the server. There are a few different ways this can happen:
+
+    * Path parameters - The dynamic values passed in your route path. e.g. `/users/:id`.
+    * Query parameters - The query string at the end of a URL after the `?` in key/value pairs. e.g. `?page=1`
+    * Form parameters - Data sent through HTTP POST. This may be formatted as JSON.
+    * Multipart parameters - Similar to form parameters, but generally to contain a file.
+
+    When parameters are passed to an action, you have access to these with the `params` method.
+
+    Here's an example of using parameters when visiting the `/users?page=1&filter=active` path:
+
+    ```crystal
+    # src/actions/users/index.cr
+    class Users::Index < BrowserAction
+
+      get "/users" do
+        filter = params.get(:filter) # type String
+        page = params.get(:page) # type String
+        per = params.get(:per) # Error! there is no parameter :per
+
+        plain_text "All users starting on page \#{page}"
+      end
+    end
+    ```
+
+    You can also use the `params.get?(:key)` method to return `nil` if the key doesn't exist.
+
+    ```crystal
+    get "/users" do
+      per = params.get?(:per) # returns nil
+
+      plain_text "..."
+    end
+    ```
+
+    > By default, all param values are trimmed of blankspace. If you need the raw value,
+    > use `params.get_raw(:key)` or `params.get_raw?(:key)`.
```

Comment:
Actually just one thought. It might be good to add a not somewhere here that it is preferred to use the type-safe macro `param` and link to it below, so people realize they can use that instead and it is preferred 99% of the time

Reply:
Ok, I think I see what you're saying. I'll need to think about how to word that.

Reply:
Alright, I moved that section to the top,and reworded a few things. I think this flows much better now. Thanks!